File: pset6/caesar.py
##
# Implements the caesar cipher
##

import cs50
import sys


def main():

    # check if called with exactly one argument
    if len(sys.argv) != 2:
        show_usage()
        exit(1)

    # check if the argument is non-negative integer
    if not sys.argv[1].isdigit():
        show_usage()
        exit(1)
    k = int(sys.argv[1])

    # ask for plaintext
    print("plaintext:  ", end="")
    p = cs50.get_string()

    # calculate and print encrypted text
    print("ciphertext: ", end="")
    for c in p:

        # encrypt and print alphabetic characters
        if c.isalpha():
            print(enc(ord(c), k), end="")

        # just print everything else
        else:
            print(c)

    # print newline for better formatting
    print()
    exit(0)


# this function encrypts single character c with a key k
# and returns the encrypted char
def enc(c, k):

    # shift sets the distance to zero for the upper and the lower case letters
    shift = 65

    # check and set for lowcase
    if c > 96:
        shift = 97
    return chr(((c - shift + k) % 26) + shift)


def show_usage():
    print("Usage: ./caesar k")

if __name__ == "__main__":
    main()

File: pset2/caesar/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

char enc(char c, int key);
void show_usage();

int main(int argc, string argv[])
{
    // check if called with exactly one argument
    if (argc != 2)
    {
        show_usage();
        return(1);
    }
    
    // get the key from the first argument
    int k = atoi(argv[1]);
    
    // check if k is non-negative integer
    if (k < 0)
    {
        show_usage();
        return(1);
    }
    
    // ask for plaintext
    printf("plaintext:  ");
    string p = get_string();

    // calculate and print encrypted text
    printf("ciphertext: ");
    for (int i = 0, l = strlen(p); i < l; i++)
    {

        // encrypt and print alphabetic characters
        if (isalpha(p[i]))
        {
            
            printf( "%c", enc(p[i],k) );
        }
        
        // just print everything else
        else
        {
            printf( "%c", p[i]);
        }
    }

    // print newline for bettter formatting
    printf("\n");
}

// this function encrypts single character c with a key k and returns the encrypted char
char enc(char c, int k)
{
    // shift sets the distance to zero for the upper and the lower case letters
    char shift = 65;
    if (islower(c))
    {
        shift = 97;
    }
    return ( ((c - shift + k) % 26) + shift  );
}

// print usage
void show_usage()
{
    printf("Usage: ./caesar k\n");
}
